OpenRA.Game/Traits/TraitsInterfaces.cs

```diff
@@ -445,7 +445,7 @@ public interface IWarhead
 		int Delay { get; }
 		bool IsValidAgainst(Actor victim, Actor firedBy);
 		bool IsValidAgainst(FrozenActor victim, Actor firedBy);
-		void DoImpact(Target target, Actor firedBy, IEnumerable<int> damageModifiers);
+		void DoImpact(Target target, Target guidedTarget, Actor firedBy, IEnumerable<int> damageModifiers);
```

Comment:
Why do we need to distinguish between `target` and `guidedTarget`?
Why can't the projectile pass the guided target as `target`?

Reply:
We still need `target` for current position, and `guidedTarget` for position/actor the spawned weapon should fire at.

Reply:
Can we implement `FireWeaponAtTargetWarhead` instead as a subclass of the `Missile` projectile, avoiding the need to update every warhead/projectile to handle this special case?

Reply:
This could also help with the
> OpenRA.Utility(1,1): Error: Weapon `quadlauncher`: projectile RangeLimit lower than weapon range!

Reply:
The GuidedTarget vs Target separation is how Ares works and how TS worked.

Reply:
I'd also like to support this for all projectile types.

Reply:
Ok. My objection is to the proliferation of arguments, so what do you think about introducing a `WarheadArgs` class to pass these instead (or can we reuse `ProjectileArgs`?).

Reply:
At first glance a `WarheadArgs` class sounds cleaner, though I need to check how many of the projectile args are useful for warheads as well (answer could be 'all').